runtime/src/gpu/common/rocm-utils.h

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2020-2023 Hewlett Packard Enterprise Development LP
+ * Copyright 2004-2019 Cray Inc.
+ * Other additional copyright holders may be indicated within.  *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __HIP_PLATFORM_AMD__
+#define __HIP_PLATFORM_AMD__
+#endif
+#include <hip/hip_common.h>
+#include <hip/hip_runtime.h>
+
+#if __has_include(<rocm-core/rocm_version.h>)  // 5.x wants this
+#include <rocm-core/rocm_version.h>
+#elif __has_include(<rocm/rocm_version.h>)  // 4.x wants this
+#include <rocm/rocm_version.h>
+#elif __has_include(<rocm_version.h>)  // Deprecated. 5.x used to want this
+#include <rocm_version.h>
+#elif !defined(ROCM_VERSION_MAJOR)
+#define ROCM_VERSION_MAJOR 4 // this is the safe bet
```

Comment:
I guess that's safe? I might just set it to 0.

Reply:
My reasoning was, we don't even know if 3 existed :) The lowest version we support is 4. Of course this doesn't tell much because we don't know how backwards compatible ROCm is... probably not too much. But right now, this is only for reductions which is not supported with 4. So, this setting disables reductions effectively. Setting it to 0, can be less future-proof if we rely on this more. But in any case, I can change it to 0 if you want me to. I prefer to keep it mostly due to inertia.

Reply:
Ah right you'd error out if you have < 4 (https://github.com/chapel-lang/chapel/blob/main/util/chplenv/chpl_gpu.py#L268). Ok yeah keeping it at 4 seems safe then.